readme.md

```diff
@@ -59,26 +59,39 @@ import "reflect-metadata";
 
 # Injectable basics
 
+## Decorators vs Registration API
+
+This library performs runtime introspection in order to determine what types it should construct.  To do this the library uses metadata and generally this metadata will be implicitly captured for you if you have enabled TypeScripts `experimentalDecorators` and a class is decorated with any decorator. However, you do not need to use decorators if you do not wish to.  In the case where no decorators are applied you will need to manually provide the metadata via the registration API.  **Note** Managing the metadata explicitly can be time consuming so we **recommend using the auto generated metadata approach by default**. 
+
 ## Creating an injectable type
 
-The easiest way to make a type injectable is to annotate it with the @Injectable annotation.  All types by default decorated in this way will be available for injection in any runtime context. 
+The easiest way to make a type injectable is to decorate it with the @Injectable decorator.  All types by default decorated in this way will be available for injection in any runtime context. 
 
 ```typescript
 import { Injectable } from '@morgan-stanley/needle';
 
 @Injectable()
-class MyThing {}
+class Pet {}
+
+@Injectable()
+class Owner {
+    constructor(pet: Pet) {}
+}
 ```
 
-While annotations are recommended for most use cases, you can also achieve the same using the Injector API. You gain access to this API by importing the `getRootInjector` function. 
+While decorators are recommended, you can also achieve the same using the Injector API. You gain access to this API by importing the `getRootInjector` function. **IMPORTANT** If you have decided not to use decorators for you injectable types, you will need to provide the constructor metadata explicitly.  Below is an example of how you can do that. 
 
 ```typescript
 import { getRootInjector } from '@morgan-stanley/needle';
 
-class MyThing {}
+class Pet {}
 
-//Equivalent to annotation
-getRootInjector().register(MyThing)
+class Owner {
+    constructor(pet: Pet) {}
+}
+
+//Equivalent to decorator
+getRootInjector().register(Owner, { metadata: Pet }).register(Pet)
```

Comment:
So, there's no type safety here. If `Pet` had constructor params you'd get a runtime error and have to add in the class of the type it needs (like `class Pet{constructor(food: Food){}}`). If `Food` had constructor params you'd get another runtime error and so on.
Is that about right?